### ISSUE TITLE
Detect mismatch of sap.app/id in manifest.json and used namespace in project

### ISSUE THREAD

**Issue by flovogt** [SAP/ui5-linter#643]
According to []() the property `sap.app/id` has to be aligned with the namespace used in the project because SAP Workzone uses this property to create the loader config / resource roots. In scenarios where projects use a separate html file with custom resource roots definition the project can be served locally but does not run when deployed to SAP Workzone.

*manifest.json*
```json
{
	"sap.app": {
		"id": "sap.ui.demo.todo",
		"type": "application"
	},
```

*Component.js*
```json
sap.ui.define([
 	"sap/ui/core/UIComponent"
], (UIComponent) => {
	"use strict";
	return UIComponent.extend("myfancynamespace.todo.Component", {
		metadata: {
			manifest: "json",
			interfaces: ["sap.ui.core.IAsyncContentCreation"],
		}
	});
});
```

## Describe the solution you'd like

UI5 linter should report and error finding and recommend to align the `sap.app/id` and the used namespace.

**Comment by boghyon** [comment]
PS: according to the linked doc, 
> [...] If the project is the app variant of an existing application, `sap.app/id` is the ID of this app variant. The ID of the underlying application is then provided in `sap.ui5/componentName`.

the `sap.ui5/componentName` value should match the _Component.js_ namespace (`"myfancynamespace.todo"`) in **app variant** projects.
